### PR TITLE
feat: Adding calls to custom style hooks derived from context in react-accordion components

### PR DESCRIPTION
**Pull request by GeoffCoxMSFT**
## Changes
- Updated components to call custom style hook

## Issues
- Updates #27058

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                                                                                             |      Baseline (minified/GZIP) |                            PR |                                                                                                                                                                                                      Change |
| :---------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: | ----------------------------: | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-accordion</samp> <br /> <abbr title='bundle-size/Accordion.fixture.js'>Accordion (including children components)</abbr>                                           |  `85.876 kB`<br />`26.019 kB` |   `88.82 kB`<br />`26.535 kB` | `2.944 kB` <br />`516 B`  |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `205.644 kB`<br />`57.056 kB` | `205.851 kB`<br />`57.021 kB` |    `207 B` <br />`-35 B`  |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                |         Size (minified/GZIP) |
| ---------------------------------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr> | `67.061 kB`<br />`18.105 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>               | `37.891 kB`<br />`12.039 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                            |   `6.324 kB`<br />`2.129 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/efb2274e37c0237f3a10a4630b2696a9880f6398'>efb2274e37c0237f3a10a4630b2696a9880f6398</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27059/perf-test-react-components/Avatar-mount_base.html">1322</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27059/perf-test-react-components/Avatar-mount.html">1323</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27059/perf-test-react-components/Button-mount_base.html">946</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27059/perf-test-react-components/Button-mount.html">949</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27059/perf-test-react-components/Field-mount_base.html">2025</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27059/perf-test-react-components/Field-mount.html">2050</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27059/perf-test-react-components/FluentProvider-mount_base.html">1660</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27059/perf-test-react-components/FluentProvider-mount.html">1628</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27059/perf-test-react-components/FluentProviderWithTheme-mount_base.html">636</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27059/perf-test-react-components/FluentProviderWithTheme-mount.html">648</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27059/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">598</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27059/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">595</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27059/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">618</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27059/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">623</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27059/perf-test-react-components/InfoButton-mount_base.html">574</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27059/perf-test-react-components/InfoButton-mount.html">565</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27059/perf-test-react-components/MakeStyles-mount_base.html">1925</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27059/perf-test-react-components/MakeStyles-mount.html">1919</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27059/perf-test-react-components/Persona-mount_base.html">3122</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27059/perf-test-react-components/Persona-mount.html">3055</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27059/perf-test-react-components/SpinButton-mount_base.html">2486</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27059/perf-test-react-components/SpinButton-mount.html">2462</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluentuiv9"/>

File: packages/react-components/react-accordion/src/components/Accordion/Accordion.tsx [hunk 1]
```diff
@@ -5,6 +5,7 @@ import { useAccordionContextValues_unstable } from './useAccordionContextValues'
 import type { AccordionProps } from './Accordion.types';
 import type { ForwardRefComponent } from '@fluentui/react-utilities';
 import { useAccordionStyles_unstable } from './useAccordionStyles';
+import { useCustomStyleHooks_unstable } from '@fluentui/react-shared-contexts';
```
Comment (khmakoto) [inline packages/react-components/react-accordion/src/components/Accordion/Accordion.tsx:8]: _nit:_ Keep `import type` statements at the bottom

```suggestion
import { useCustomStyleHooks_unstable } from '@fluentui/react-shared-contexts';
import { useAccordionStyles_unstable } from './useAccordionStyles';
import type { AccordionProps } from './Accordion.types';
import type { ForwardRefComponent } from '@fluentui/react-utilities';
```

File: packages/react-components/react-accordion/src/components/AccordionItem/AccordionItem.tsx [hunk 2]
```diff
@@ -5,6 +5,7 @@ import { renderAccordionItem_unstable } from './renderAccordionItem';
 import type { AccordionItemProps } from './AccordionItem.types';
 import type { ForwardRefComponent } from '@fluentui/react-utilities';
 import { useAccordionItemStyles_unstable } from './useAccordionItemStyles';
+import { useCustomStyleHooks_unstable } from '@fluentui/react-shared-contexts';
```
Comment (khmakoto) [inline packages/react-components/react-accordion/src/components/AccordionItem/AccordionItem.tsx:8]: _nit:_ Keep `import type` statements at the bottom

```suggestion
import { useCustomStyleHooks_unstable } from '@fluentui/react-shared-contexts';
import { useAccordionItemStyles_unstable } from './useAccordionItemStyles';
import type { AccordionItemProps } from './AccordionItem.types';
import type { ForwardRefComponent } from '@fluentui/react-utilities';
```

File: packages/react-components/react-accordion/src/components/AccordionPanel/AccordionPanel.tsx [hunk 3]
```diff
@@ -4,6 +4,7 @@ import { renderAccordionPanel_unstable } from './renderAccordionPanel';
 import { useAccordionPanelStyles_unstable } from './useAccordionPanelStyles';
 import type { AccordionPanelProps } from './AccordionPanel.types';
 import type { ForwardRefComponent } from '@fluentui/react-utilities';
+import { useCustomStyleHooks_unstable } from '@fluentui/react-shared-contexts';
```
Comment (khmakoto) [inline packages/react-components/react-accordion/src/components/AccordionPanel/AccordionPanel.tsx:7]: _nit:_ Keep `import type` statements at the bottom

```suggestion
import { useCustomStyleHooks_unstable } from '@fluentui/react-shared-contexts';
import { useAccordionPanelStyles_unstable } from './useAccordionPanelStyles';
import type { AccordionPanelProps } from './AccordionPanel.types';
import type { ForwardRefComponent } from '@fluentui/react-utilities';
```

**Review by khmakoto — APPROVED** [review]

**Pull request merged**